wordpress-mobile/WordPress-Android
# Enable Search for Self-Hosted Sites

Fixes #12058 
Prerequisite [WordPress-FluxC-Android #1646](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1646)

Description
-----
This feature enables search for self-hosted sites, something that was previously only available for WordPress.com/Jetpack sites. See issue above for more info.

In general the solution is split into two parts:
1. Make the `search icon` available for when the user navigates to a self-hosted site, just as it is available for non self-hosted sites.
2. Add the user's `search query` to the XML RPC descriptor that is responsible to later make an API request for the self-hosted site. Note that with the `develop` version of this repo the `search query` argument is not available since a new version of **Android FluxC** is required (see `To test` section below).

So, the above work is closely related to the [[Feature] Add Search Filter for XML RPC Requests #1646](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1646) PR, which needs to be merge first.

Blog Posts for Self-Hosted Site (Before)
-----
<img src="https://user-images.githubusercontent.com/9729923/88916567-f7920980-d26e-11ea-9076-5779dc2ff9fb.png" width="250px" />

Blog Posts for Self-Hosted Site (After)
-----

<img src="https://user-images.githubusercontent.com/9729923/88916578-feb91780-d26e-11ea-9705-b1cd57b9c0ee.png" width="250px" />

Test:
-----
1. Build the `WordPress` app with a `fluxcVersion` of `e32d20633e68ca1fba1726b75463aa7219ca88f0` (update that within the project's parent `build.gradle` file)
1. Sign in to the app and add a self-hosted site
1. Navigate to `Blog Posts`
1. Verify that the `Search Icon` is available on the top right
1. Click on the `Search Icon` and verify the search functionality works as expected and in general works as it used to work for non self-hosted site
1. The above means that search is enabled for self-hosted sites

While testing note that search in general, for both for self-hosted and non self-hosted sites has a configuration change related bug, see #12561. However, it is out of the scope of this task to fix this bug, but it will be considered as part of another task, later on in the future.

PR submission checklist:
-----
